File: examples/comparison/comptime_table/bench.cpp
#include <cstdio>
#include <cstdint>

namespace {
constexpr int TABLE_LEN = 64;
constexpr int BUILD_ITERS = 8000;
constexpr int SUM_ROUNDS = 8;
constexpr int64_t MOD = 1000000007LL;

int64_t mix(int64_t n) {
    int64_t a = n * 100003LL;
    int64_t b = n * n;
    return (a + b * 31 + 997) % MOD;
}
}  // namespace

int main() {
    int64_t table[TABLE_LEN];
    for (int i = 0; i < TABLE_LEN; ++i) {
        int64_t v = 0;
        for (int k = 0; k < BUILD_ITERS; ++k) {
            v = (v + mix(i + k)) % MOD;
        }
        table[i] = v;
    }
    int64_t acc = 0;
    for (int r = 0; r < SUM_ROUNDS; ++r) {
        for (int j = 0; j < TABLE_LEN; ++j) {
            acc = (acc + table[j]) % MOD;
        }
    }
    std::printf("%lld\n", static_cast<long long>(acc));
    return 0;
}

File: examples/comparison/comptime_table/bench.go
package main

import (
	"fmt"
	"runtime"
)

const (
	tableLen    = 64
	buildIters  = 8000
	sumRounds   = 8
	mod   int64 = 1000000007
)

func mix(n int64) int64 {
	a := n * 100003
	b := n * n
	return (a + b*31 + 997) % mod
}

func main() {
	table := make([]int64, tableLen)
	for i := 0; i < tableLen; i++ {
		var v int64
		for k := 0; k < buildIters; k++ {
			v = (v + mix(int64(i+k))) % mod
		}
		table[i] = v
	}
	var acc int64
	for r := 0; r < sumRounds; r++ {
		for j := 0; j < tableLen; j++ {
			acc = (acc + table[j]) % mod
		}
	}
	runtime.KeepAlive(acc)
	fmt.Println(acc)
}
